packages/material-ui-unstyled/src/BadgeUnstyled/BadgeUnstyled.js

```diff
@@ -3,33 +3,25 @@ import PropTypes from 'prop-types';
 import clsx from 'clsx';
 import { unstable_capitalize as capitalize, usePreviousProps } from '@material-ui/utils';
 import isHostComponent from '../utils/isHostComponent';
-import badgeUnstyledClasses, { getBadgeUtilityClass } from './badgeUnstyledClasses';
+import composeClasses from '../composeClasses';
+import { getBadgeUtilityClass } from './badgeUnstyledClasses';
 
 const useUtilityClasses = (styleProps) => {
-  const { variant, anchorOrigin, overlap, invisible, classes = {} } = styleProps;
-
-  return {
-    root: clsx(badgeUnstyledClasses.root, classes.root),
-    badge: clsx(
-      badgeUnstyledClasses.badge,
-      classes.badge,
-      getBadgeUtilityClass(variant),
-      badgeUnstyledClasses[
-        `anchorOrigin${capitalize(anchorOrigin.vertical)}${capitalize(
-          anchorOrigin.horizontal,
-        )}${capitalize(overlap)}`
-      ],
-      classes[
-        `anchorOrigin${capitalize(anchorOrigin.vertical)}${capitalize(
-          anchorOrigin.horizontal,
-        )}${capitalize(overlap)}`
-      ],
-      {
-        [badgeUnstyledClasses.invisible]: invisible,
-        [classes.invisible]: invisible,
-      },
-    ),
+  const { variant, anchorOrigin, overlap, invisible, classes } = styleProps;
+
+  const slots = {
+    root: ['root'],
+    badge: [
+      'badge',
+      variant, // there was only utility class for the variant previously
```

Comment:
```suggestion
      variant,
```
I'd say we are fixing issue with this.